src/Microsoft.ML.Data/Prediction/CalibratorCatalog.cs

```diff
@@ -425,26 +419,129 @@ public sealed class PavCalibratorEstimator : CalibratorEstimatorBase<PavCalibrat
 
         [BestFriend]
         private protected override CalibratorTransformer<PavCalibrator> Create(IHostEnvironment env, PavCalibrator calibrator)
-            => new PavCalibratorTransformer(env, calibrator);
+            => new PairAdjacentViolatorsCalibratorTransformer(env, calibrator);
 
     }
 
     /// <summary>
-    /// The <see cref="ITransformer"/> implementation obtained by training a <see cref="PavCalibratorEstimator"/>
+    /// The <see cref="ITransformer"/> implementation obtained by training a <see cref="PairAdjacentViolatorsCalibratorEstimator"/>
     /// </summary>
-    public sealed class PavCalibratorTransformer : CalibratorTransformer<PavCalibrator>
+    public sealed class PairAdjacentViolatorsCalibratorTransformer : CalibratorTransformer<PavCalibrator>
     {
         internal const string LoadName = "PavCalibratTransf";
 
-        internal PavCalibratorTransformer(IHostEnvironment env, PavCalibrator calibrator)
+        internal PairAdjacentViolatorsCalibratorTransformer(IHostEnvironment env, PavCalibrator calibrator)
           : base(env, calibrator, LoadName)
         {
         }
 
         // Factory method for SignatureLoadModel.
-        private PavCalibratorTransformer(IHostEnvironment env, ModelLoadContext ctx)
+        private PairAdjacentViolatorsCalibratorTransformer(IHostEnvironment env, ModelLoadContext ctx)
             : base(env, ctx, LoadName)
         {
         }
     }
 }
+
+namespace Microsoft.ML
+{
+    public static class CalibratorsCatalog
```

Comment:
>CalibratorsCatalog [](start = 24, length = 18)

So, maybe we can just have `CalibratorsCatalog` be a class, with actual methods, that is, not extension methods, given that it's in the same assembly. We sometimes do extension methods, but we often also put methods directly on the thing when the "scope" of what is to be done is fairly limited, or there isn't some great question of broader consistency, or somesuch.

Reply:
by saying that you mean something like `mlContext.Calirbrators.Platt()` or you mean `CalibratorsCatalog.Platt(MlContext context, ...)`
Second feels outlier to way how we currently handle everything else.

Reply:
No. I am saying that these should be actual methods on the catalog class itself.

Reply:
Let me be more explicit. I could write the following code:

```chsarp
public class Foo { }

public static class FooExtensions
{
    public static void Bar(this Foo foo, ...) { ... }
    public static void Biz(this Foo foo, ...) { ... }
    public static void Blam(this Foo foo, ...) { ... }
}
```

As in fact we do when we want to have separation between multiple things. But, in this particular case, *everything* we'd want to do with this catalog lives in the same assembly where the assembly is. So I could have written the following.

```chsarp
public class Foo
{
    public void Bar(...) { ... }
    public void Biz(...) { ... }
    public void Blam(...) { ... }
}
```

I would argue the second is easier and more clear. I understand the reasons why we sometimes phrase these catalog operations as extension methods, what I'm saying is, in this case, it is not necessary or helpful to do so. If you disagree that's fine I guess, it just seems strange.

Reply:
Thank you. I'm sorry you had to write down this explanation for me. I've been slightly tired lately and with all other distractions it looks like I lost ability to focus and understand other people.

Reply:
Ha ha, that's fine Ivan. I remember yesterday when I was so exhausted and sick I confused the heck out of people with what were (if I look back on them) some insane ramblings.